src/plugins/explore/public/application/legacy/discover/application/components/visualizations/line/line_chart_utils.ts

```diff
@@ -0,0 +1,274 @@
+    case 'dashed':
+      return [5, 5];
+    case 'dot-dashed':
+      return [5, 5, 1, 5];
+    case 'full':
+    default:
+      return undefined;
+  }
+};
+
+/**
+ * Get Vega interpolation from UI lineMode
+ * @param lineMode The line mode ('straight', 'smooth', or 'stepped')
+ * @returns The corresponding Vega interpolation value
+ */
+export const getVegaInterpolation = (lineMode: string): string => {
+  switch (lineMode) {
+    case 'straight':
+      return 'linear';
+    case 'smooth':
+      return 'monotone';
+    case 'stepped':
+      return 'step-after';
+    default:
+      return 'monotone';
+  }
+};
+
+/**
+ * Build proper Vega-Lite mark configuration
+ * @param styles The style options
+ * @param markType The mark type ('line' or 'bar')
+ * @returns The mark configuration object
+ */
+export const buildMarkConfig = (
+  styles: Partial<LineChartStyleControls> | undefined,
+  markType: 'line' | 'bar' = 'line'
+): any => {
+  // Default values - handle undefined styles object
+  const showLine = styles?.showLine !== false;
+  const showDots = styles?.showDots !== false;
+  const lineWidth = styles?.lineWidth ?? 2;
+  const lineMode = styles?.lineMode ?? 'smooth';
+  const addTooltip = styles?.addTooltip !== false;
+
+  if (markType === 'bar') {
+    return {
+      type: 'bar',
+      opacity: 0.7,
+      tooltip: addTooltip,
+    };
+  }
+
+  // For line charts
+  if (!showLine && showDots) {
+    // Only show points
+    return {
+      type: 'point',
+      tooltip: addTooltip,
+      size: 100,
+    };
+  } else if (showLine && !showDots) {
+    // Only show line
+    return {
+      type: 'line',
+      tooltip: addTooltip,
+      strokeWidth: lineWidth,
+      interpolate: getVegaInterpolation(lineMode),
+    };
+  } else if (showLine && showDots) {
+    // Show both line and points
+    return {
+      type: 'line',
+      point: true,
+      tooltip: addTooltip,
+      strokeWidth: lineWidth,
+      interpolate: getVegaInterpolation(lineMode),
+    };
+  } else {
+    // Neither line nor dots - fallback to points
+    return {
+      type: 'point',
+      tooltip: addTooltip,
+      size: 0, // Make points invisible
+    };
+  }
+};
+
+/**
+ * Create threshold line layer
+ * @param styles The style options
+ * @returns The threshold layer configuration or null if disabled
+ */
+export const createThresholdLayer = (styles: Partial<LineChartStyleControls> | undefined): any => {
+  if (!styles?.thresholdLine?.show) {
+    return null;
+  }
+
+  const thresholdLayer: any = {
+    mark: {
+      type: 'rule',
+      color: styles.thresholdLine.color,
+      strokeWidth: styles.thresholdLine.width,
+      strokeDash: getStrokeDash(styles.thresholdLine.style),
+      tooltip: styles.addTooltip !== false,
+    },
+    encoding: {
+      y: {
+        datum: styles.thresholdLine.value,
+        type: 'quantitative',
+      },
+    },
+  };
+
+  // Add tooltip content if enabled
+  if (styles.addTooltip !== false) {
```

Comment:
nitpick: `if (styles.addTooltip)`

Reply:
i only check against false; since we want default behavior is to have tooltip. so if addTooltip is undefined, we enable it